### PR TITLE
Added a hands-on self-containted MLflow/Ray Serve deployment example

### PR DESCRIPTION
**Pull request by dmatrix**
Signed-off-by: Jules S.Damji <jules@anyscale.com>

## Why are these changes needed?

To augment examples in Ray Serve's [Integration with Model Registries](https://github.com/ray-project/ray/compare/master...dmatrix:br_jsd_add_mlflow_serve_example?expand=1#id4) section,
I added a hands-on self-contained example of how to use `mlflow.sklearn.autolog()`, which logs
all the metrics, parameters, and artifacts to the local model registry; and a deployment class that
loads the registered model as a `pyfunc` model and does the prediction.

This complete example augments the previous partial code snippet. 

## Checks

- [x ] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]
Looks great, thanks for adding this!  I'll make a followup issue to run this code sample in CI using Sphinx's `literalinclude`.  I'll need to figure out where to install `mlflow` in the CI setup but it should be straightforward.

File: doc/source/serve/ml-models.rst [hunk 1]
```diff
@@ -114,6 +114,155 @@ To serve multiple different MLflow models in the same program, use the ``name``
   The above approach will work for any model registry, not just MLflow.
   Namely, load the model from the registry in ``__init__``, and forward the request to the model in ``__call__``.
 
+For a complete hands-on and seamless integration with MLflow, try this self-contained example on your laptop.
+But first install `mlflow`.
```
Comment (architkulkarni) [inline doc/source/serve/ml-models.rst:118]:
```suggestion
But first install ``mlflow``.
```

File: doc/source/serve/ml-models.rst [hunk 2]
```diff
@@ -114,6 +114,155 @@ To serve multiple different MLflow models in the same program, use the ``name``
...
+            if not os.path.exists(local_dir):
+                os.mkdir(local_dir)
+            client = MlflowClient()
+            local_path = f"{client.download_artifacts(run_id, 'labels', local_dir)}/iris_labels.json"
+            with open(local_path, "r") as f:
+                self.label_list = json.load(f)
+
+        async def __call__(self, startlette_request):
+            payload = await startlette_request.json()
+            print(f"Worker: received starlet request with data: {payload}")
```
Comment (architkulkarni) [inline doc/source/serve/ml-models.rst:198]:
```suggestion
        async def __call__(self, starlette_request):
            payload = await starlette_request.json()
            print(f"Worker: received Starlette request with data: {payload}")
```

File: doc/source/serve/ml-models.rst (hunk 1 above)
Comment (dmatrix) [reply doc/source/serve/ml-models.rst]: Incorporated the feedback

File: doc/source/serve/ml-models.rst (hunk 2 above)
Comment (dmatrix) [reply doc/source/serve/ml-models.rst]: Made the changes.

**Pull request merged**